Fix `Activate this design` button for Marketplace themes

gcsecsey opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Closes https://github.com/Automattic/dotcom-forge/issues/3581

## Proposed Changes

While the AT transfer is running and the progress bar is shown, the progress is being polled from the back end. Calypso also stores the `is_wpcom_atomic` flag in state as part of the cached response from the `/rest/v1.2/sites/:siteId` endpoint.  When the AT transfer is complete, and the user is redirected to the thank you page, this data is stale and needs to be refreshed.

This PR adds a query to refresh site details on the thank you page.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* with a SIMPLE site purchase the Business plan, or register a new test site from `/start/business`
* purchase a partner theme which has plugin dependencies (eg. `themes/makoney` or `themes/nokul`)
* after checkout, wait for the AT transfer to finish
* on the Thank You page, click the `Activate this theme` button
* check that the ActivationModal is shown (the eligibility modal before should **NOT** be shown)
   <img width="1892" alt="CleanShot 2023-09-01 at 13 42 08@2x" src="https://github.com/Automattic/wp-calypso/assets/11555574/d787dda7-3d66-4ead-9c45-67dc84ce8671">
* check that the theme is activated

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

gcsecsey committed:
add eligibility modal to thank you theme section

gcsecsey committed:
add query sites

gcsecsey committed:
revert debug changes

epeicher reviewed (changes requested):
The code changes make sense to me, but if I clear the cache and follow the testing instructions, I can still reproduce the issue described in the linked task. 

Please see a video with the steps:

https://github.com/Automattic/wp-calypso/assets/3519124/7f69a7ad-6936-460a-8d56-401571060921

In the end, the theme is installed but not active, and the dependent plugins are not installed:
|Themes | Installed plugins|
|-------|------|
|![CleanShot 2023-09-05 at 10 06 13@2x](https://github.com/Automattic/wp-calypso/assets/3519124/0895a565-7b41-4f5a-9253-2da80c7df8f3)| ![CleanShot 2023-09-05 at 10 06 48@2x](https://github.com/Automattic/wp-calypso/assets/3519124/e75ba8dc-8a51-4682-ad9b-6ecb42706ba8)|

gcsecsey commented:
> The code changes make sense to me, but if I clear the cache and follow the testing instructions, I can still reproduce the issue described in the linked task.

Thanks @epeicher for testing this, I'll investigate running this with an empty cache further. Maybe the timing of the refetch is still too soon in this case.

> In the end, the theme is installed but not active, and the dependent plugins are not installed:

I've also tested this again with both themes and found an interesting detail. When testing with the Makoney theme, the dependency plugin `AinoBlocks` is installed. When testing with the Nokul theme, the dependencies aren't installed either.

Maybe there's also some difference in the configuration of these themes?

Makoney:
<img width="391" alt="CleanShot 2023-09-05 at 11 15 17@2x" src="https://github.com/Automattic/wp-calypso/assets/11555574/d095e966-7254-46ff-9b93-f4ed359d8ab1">

Nokul:
<img width="458" alt="CleanShot 2023-09-05 at 11 14 52@2x" src="https://github.com/Automattic/wp-calypso/assets/11555574/5b0a3c46-7bbc-4c7f-88c3-15c6be14c4c1">

epeicher commented:
> I've also tested this again with both themes and found an interesting detail. When testing with the Makoney theme, the dependency plugin `AinoBlocks` is installed. When testing with the Nokul theme, the dependencies aren't installed either.
> 
> Maybe there's also some difference in the configuration of these themes?

One difference I've found that may or may not be related is that `Makoney` only has one dependency but `Nokul` has two dependencies. I don't know if that could be related or if it just affects the timing 🤷 . Thanks for investigating this further!

gcsecsey committed:
remove eligibilitywarningmodal from thank you page

gcsecsey committed:
Use the isSiteAtomic hook for checking transfer completion

gcsecsey commented:
I added the `isSiteAtomic` hook to the `useAtomicTransfer` hook as a final check. This makes the progress bar shown up until the `is_wpcom_atomic` flag is returned as true from the sites endpoint. This guarantees that the Calypso state is up-to-date when the user can click the `Activate` button. cc @WBerredo 

I couldn't find the root cause yet of why the dependency plugins of the `Nokul` theme. I'll raise a separate PR to address that, for now I've removed checking the plugins from the test steps. 

@epeicher please give another review to this when you can. 🙏

epeicher reviewed (approved):
I have tested and reviewed the changes and they Look Good To Me! 🚢 

https://github.com/Automattic/wp-calypso/assets/3519124/b8431d41-8cd4-415d-a5e8-485963eadcc9

epeicher commented:
>I couldn't find the root cause yet of why the dependency plugins of the Nokul theme. I'll raise a separate PR to address that, for now I've removed checking the plugins from the test steps.

That's great as it could be unrelated, so we can fix this issue now and iterate on that.
